src/Knp/DictionaryBundle/Form/Type/DictionaryType.php

```diff
@@ -25,31 +25,29 @@ public function __construct(DictionaryRegistry $registry)
     /**
      * {@inheritdoc}
      */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    public function configureOptions(OptionsResolver $resolver)
     {
-        $choices = function (Options $options) {
+        $registry = $this->registry;
+
+        $choices = function (Options $options) use ($registry) {
             $name    = $options['name'];
-            $choices = $this->registry[$name]->getValues();
+            $choices = $registry[$name]->getValues();
 
             return $choices;
         };
 
-        $resolver->setDefaults(array('name' => null, 'choices' => $choices));
+        $resolver
+            ->setDefault('choices', $choices)
+            ->setRequired(array('name'))
+            ->setAllowedValues('name', array_keys($this->registry->all()))
+        ;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getParent()
     {
-        return 'choice';
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getName()
-    {
-        return 'dictionary';
+        return 'Symfony\Component\Form\Extension\Core\Type\ChoiceType';
```

Comment:
We still want the bundle to be PHP 5.3 compatible (or at least 5.4, it's still being discussed)